### PR TITLE
Put metric before label value in error message when 'label value too long' 

### LINKED ISSUE #1582: Error message for 'label value too long' misses important information
The error seen by sender in Prometheus logs is like this:

```
level=warn ts=2019-07-22T09:25:16.899437956Z caller=queue_manager.go:530 component=remote queue=0:https://cloud.weave.works./api/prom/push msg="Error sending samples to remote storage" count=1000 err="server returned HTTP status 400 Bad Request: label value too long: \"{\\\"apiVersion\\\":\\\"extensions/v1beta1\\\",\\\"kind\\\":\\\"DaemonSet\\\",\\\"metadata\\\":{\\\"annotations\\\":{\\\"flux.weave.works/sync-checksum\\\":\\\"0b6f1382edb3f01d6a64cd775b46008b0f61003e\\\"},\\\"labels\\\":{\\\"flux.weave.works/sync-gc-mark\\\":\" metric \"kub"
level=warn ts=2019-07-22T09:25:32.00067315Z caller=queue_manager.go:530 component=remote queue=0:https://cloud.weave.works./api/prom/push msg="Error sending samples to remote storage" count=1000 err="server returned HTTP status 400 Bad Request: label value too long: \"{\\\"apiVersion\\\":\\\"extensions/v1beta1\\\",\\\"kind\\\":\\\"Deployment\\\",\\\"metadata\\\":{\\\"annotations\\\":{\\\"flux.weave.works/automated\\\":\\\"true\\\",\\\"flux.weave.works/automated_msg\\\":\\\"test\\\",\\\"flux.weave.works/automated_user\\\":\\\"Maarten\" metric \""
```

It looks like something is truncating the message just as it gets to the `metric=...` part.  Maybe put that before the value?
And it would also be useful to state the label name.

### PR DESCRIPTION
**Pull request by ChinYing-Li**
(#1582)

Signed-off-by: ChinYing-Li <chinying.li@mail.utoronto.ca>

**What this PR does**:
Swap the position of the metric string and the label value string in error message of `labelValueTooLong`.
Now the metric string is placed before the  label value string.

**Which issue(s) this PR fixes**:
Fixes #1582

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
   * `-alertmanager.cluster.advertise-address` instead of `-cluster.advertise-address`
   * `-alertmanager.cluster.peers` instead of `-cluster.peer`
   * `-alertmanager.cluster.peer-timeout` instead of `-cluster.peer-timeout`
+* [CHANGE] Put metric before label value in error message when `labelValueTooLong`. #3982
```
Comment (pracucci) [inline CHANGELOG.md:13]:
```suggestion
* [ENHANCEMENT] Put metric before label value in the "label value too long" error message. #3982
```

File: pkg/util/validation/validate.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ const (
 	errInvalidMetricName  = "sample invalid metric name: %.200q"
 	errInvalidLabel       = "sample invalid label: %.200q metric %.200q"
 	errLabelNameTooLong   = "label name too long: %.200q metric %.200q"
-	errLabelValueTooLong  = "label value too long: %.200q metric %.200q"
+	errLabelValueTooLong  = "label value too long: metric %.200q value %.200q"
```
Comment (pracucci) [inline pkg/util/validation/validate.go:38]:
```suggestion
	errLabelValueTooLong  = "label value too long for metric %.200q label name: %.200q"
```

**Review by pracucci — COMMENTED** [review]
Thanks for working on this! LGTM (modulo a couple of nits)

**Comment by pracucci** [comment]
@ChinYing-Li Sorry to bother you. I've just merged #3990 which impacts on this PR. Could you apply the same change (picking up the suggestion I gave above) to the error which is now defined in `pkg/util/validation/errors.go`, please?

**Comment by ChinYing-Li** [comment]
Given the refactor, now there's no simple way to identify whether s `genericValidationError` is a `label value too long` error, hence unable to make  `label value too long` a special case in `func (e *genericValidationError) Error() string`.  Introducing another member e.g. `errorType` to `genericValidationError` would be against the design; do you have any suggestion on how to approach this? @pracucci

**Comment by pstibrany** [comment]
> Given the refactor, now there's no simple way to identify whether s `genericValidationError` is a `label value too long` error, hence unable to make `label value too long` a special case in `func (e *genericValidationError) Error() string`. Introducing another member e.g. `errorType` to `genericValidationError` would be against the design; do you have any suggestion on how to approach this? @pracucci

I think the solution is to not use `genericValidationError` for "label value too long", but define custom error.

**Comment by pracucci** [comment]
> I think the solution is to not use genericValidationError for "label value too long", but define custom error.

Yes, I agree.

**Comment by ChinYing-Li** [comment]
Closing this PR. Relevant development done based on recent changes is at #4018.

**Pull request closed without merging**